src/util.c

```diff
@@ -121,3 +121,85 @@ void take_screenshot(bool verbose)
 
 	return;
 }
+
+/*	https://stackoverflow.com/a/37132824
+*/
+#ifdef __linux
+	#if _POSIX_C_SOURCE >= 200809L
+		#define sf_strdup(x) strdup(x)
+	#else
+
+static char *sf_strdup(char *src)
```

Comment:
I'm not familiar with `sf_strdup`, but a quick lookup suggests that it `abort`s on error (namely memory exhaustion), whereas this implementation can potentially return `NULL`.

You could fix this by using `xmalloc` or another fail-early `malloc` variant, but I'd prefer we just use plain-old `strdup` instead (and do relevant `NULL` checks).

Reply:
I'd prefer to avoid an own implementation of strdup as well. To use regular `strdup()` you need to define `_POSIX_C_SOURCE` as `200809L` in the Makefile, at least for Linux. Is that okay?

Reply:
Yep, that's fine by me. Change it for every platform, though, not just Linux.